pkg/collector/corechecks/cluster/orchestrator/discovery/collector_discovery.go

```diff
@@ -27,35 +28,107 @@ const (
 	defaultAPIServerTimeout = 20 * time.Second
 )
 
+type discoveryCache struct {
```

Comment:
The cache is filled once during crd discovery and then re-used whenever we add a new crd. 
I am still debating to move this to a dedicated file (which I probably will)